src/app/functions/server/onerep.ts

```diff
@@ -0,0 +1,283 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+import AppConstants from "../../../appConstants.js";
+import mozlog from "../../../utils/log.js";
+import {
+  E164PhoneNumberString,
+  ISO8601DateString,
+  parseE164PhoneNumber,
+  parseIso8601Datetime,
+} from "../../../utils/parse.js";
+import { StateAbbr, usStates } from "../../../utils/states.js";
+const log = mozlog("external.onerep");
+
+export type ProfileData = {
+  first_name: string;
+  last_name: string;
+  city: string;
+  state: StateAbbr;
+  birth_date?: ISO8601DateString;
+  age?: ISO8601DateString;
+  phone_number?: E164PhoneNumberString;
+};
+export type CreateScanResponse = {
+  id: number;
+  profile_id: number;
+  status: "in_progress";
+  reason: "manual";
+  created_at: ISO8601DateString;
+  updated_at: ISO8601DateString;
+};
+export type OneRepMeta = {
+  current_page: number;
+  from: number;
+  last_page: number;
+  per_page: number;
+  to: number;
+  total: number;
+};
+export type Scan = {
+  id: number;
+  profile_id: number;
+  status: "in_progress" | "finished";
+  reason: "initial" | "monitoring" | "manual";
+};
+export type ListScansResponse = {
+  meta: OneRepMeta;
+  data: Scan[];
+};
+export type ScanResult = {
+  id: number;
+  url: string;
+  link: string;
+  profile_id: number;
+  first_name: string;
+  last_name: string;
+  middle_name: string;
+  age: string;
+  addresses: Array<{
+    city: string;
+    state: StateAbbr;
+    street?: string;
+    zip?: string;
+  }>;
+  relatives: string[];
+  phones: string[];
+  emails: string[];
+  data_broker: string;
+  status: RemovalStatus;
+  data_broker_id: number;
+  created_at: ISO8601DateString;
+  updated_at: ISO8601DateString;
+};
+export type ListScanResultsResponse = {
+  meta: OneRepMeta;
+  data: ScanResult[];
+};
+export type RemovalStatus =
+  | "new"
+  | "optout_in_progress"
+  | "waiting_for_verification"
+  | "removed";
+
+async function onerepFetch(
+  path: string,
+  options: Parameters<typeof fetch>[1] = {}
+) {
+  const url = "https://api.onerep.com" + path;
+  const headers = new Headers(options.headers);
```

Comment:
perhaps this part (setting header for bearer token) can be stripped out into an util thing? 
We have a corresponding getter in `/app/api/utils/auth.ts`